clang-tidy/mesos/RedundantGetCheck.cpp

```diff
@@ -0,0 +1,59 @@
+//===--- RedundantGetCheck.cpp - clang-tidy--------------------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "RedundantGetCheck.h"
+#include "clang/AST/ASTContext.h"
+#include "clang/ASTMatchers/ASTMatchFinder.h"
+
+using namespace clang::ast_matchers;
+
+namespace clang {
+namespace tidy {
+namespace mesos {
+
+void RedundantGetCheck::registerMatchers(MatchFinder *Finder) {
+  Finder->addMatcher(
+      memberExpr(has(ignoringImpCasts(
+                     cxxMemberCallExpr(
+                         callee(cxxMethodDecl(hasName("get"))),
+                         on(expr(hasType(cxxRecordDecl(
+                                     anyOf(hasName("Future"), hasName("Option"),
+                                           hasName("Try"), hasName("Result")))))
```

Comment:
Shouldn't this be `::Future`, etc., to not accidentally match classes in other namespaces?

Reply:
This doesn't trigger in Mesos. I tightened the checker nevertheless since it is good documentation.